planet/clients/orders.py

```diff
@@ -319,7 +319,10 @@ async def download_order(self,
     def _get_order_locations(order):
         links = order['_links']
         results = links.get('results', None)
-        return list(r['location'] for r in results if r)
+        try:
```

Comment:
Yeah, this would probably do it! However, what do you think about raising a ClientError exception if no links are found (i.e., L323)?

Reply:
hmmm... that is a great question. Let's think about usage here. For `_get_order_locations` it doesn't really seem to be an error scenario if there are no results - there just are no locations either (hence the empty list being returned). Maybe `download_orders` would be concerned about that but if locations is an empty list, it doesn't download any files and `filenames` is an empty list. So, it returns an empty list if no links are found. Is it considered an error if a user runs `download_order` and the order has no links, so no files were downloaded? Should that be a scenario that causes the program to halt or requires catching the exception? I'm quite open to dialogue. My initial thinking is that it shouldn't raise an exception since the user might not mind an order being empty, but it should raise a warning. Maybe that's the least helpful approach, but it's also the least heavy-handed. What are your thoughts?

Reply:
That's a good point!

Let's not raise an exception, because you're right, it should still continue to run even if no locations are provided. So, instead of raising an exception I'll return an empty list and warn the user via `LOGGER.warning()` that there were no locations found and that nothing will be downloaded.

Reply:
sounds great! and if we find we want to change our minds later, it will be easy to do so. Thanks for going through that thought exercise with me

Reply:
Alrighty, I'll leave it as a warning for now and we can change our mind later, if we wish :)